### PR TITLE
dwm-git: 20180602 -> 20200303

### PR DESCRIPTION
**Pull request by ivankovnatsky**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ivankovnatsky** [comment]
```console
nixpkgs-review pr 108150

1 package built:
dwm-git
```

File: pkgs/applications/window-managers/dwm/git.nix [hunk 1]
```diff
@@ -1,16 +1,18 @@
-{ stdenv, fetchgit, libX11, libXinerama, libXft, patches ? [], conf ? null }:
+{ stdenv, fetchgit, libX11, libXinerama, libXft, writeText, patches ? [ ]
+, conf ? null }:
 
-let
-  name = "dwm-git-20180602";
-in
+with stdenv.lib;
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/dwm/git.nix:4]: Pelase add the with where you need it and don't do it over the entire file.

File: pkgs/applications/window-managers/dwm/git.nix [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+in stdenv.mkDerivation {
   inherit name;
 
   src = fetchgit {
     url = "git://git.suckless.org/dwm";
-    rev = "b69c870a3076d78ab595ed1cd4b41cf6b03b2610";
-    sha256 = "10i079h79l4gdch1qy2vrrb2xxxkgkjmgphr5r9a75jbbagwvz0k";
+    rev = "61bb8b2241d4db08bea4261c82e27cd9797099e7";
+    sha256 = "1j3vly8dln35vnwnwwlaa8ql9fmnlmrv43jcyc8dbfhfxiw6f34l";
+
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/dwm/git.nix:15]:
```suggestion
```

File: pkgs/applications/window-managers/dwm/git.nix [hunk 3]
```diff
@@ -1,16 +1,18 @@
...
-let
-  name = "dwm-git-20180602";
-in
+with stdenv.lib;
 
-stdenv.mkDerivation {
+let name = "dwm-git-20200303";
+
+in stdenv.mkDerivation {
   inherit name;
```
Comment (SuperSandro2000) [inline pkgs/applications/window-managers/dwm/git.nix:9]:
```suggestion
stdenv.mkDerivation {
  pname = "dwm-git";
  version = "20200303";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review. If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108150` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>dwm-git</li>
</ul>
</details>

Please do not overwrite the buildPhase with make. make is automatically executed.

**Comment by SuperSandro2000** [comment]
Please squash the suggestions together.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 108150` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dwm-git</li>
  </ul>
</details>

File: pkgs/applications/window-managers/dwm/git.nix (hunk 1 above)
Comment (ivankovnatsky) [reply pkgs/applications/window-managers/dwm/git.nix]: Sandro, got it, updated.

**Comment by ivankovnatsky** [comment]
should be good, i see that gh shows some requested changes, but it should be already resolved.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108150` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>dwm-git</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108150` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>dwm-git</li>
</ul>
</details>

**Review by SuperSandro2000 — APPROVED** [review]

**Review by cole-h — APPROVED** [review]
Diff LGTM, thanks!

---

Result of `nixpkgs-review pr 108150` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dwm-git</li>
  </ul>
</details>

**Pull request merged**